pkg/kubelet/volumemanager/cache/desired_state_of_world.go

```diff
@@ -278,6 +298,9 @@ func (dsw *desiredStateOfWorld) DeletePodFromVolume(
 		// Delete volume if no child pods left
 		delete(dsw.volumesToMount, volumeName)
 	}
+
+	// create metric for volume unmount
+	dsw.volumesToUnmount[volumeName] = metrics.VolumeManagerOperationCompleteHook(volumePluginName, "volume_unmount")
```

Comment:
Do we need to add pod name as a dimension? As this umount operation is just umount volume from pod, not umount the volume device from node.

Reply:
We generally never want to add pod names as a dimension because in a large enough cluster, it will cause blowing up of cardinality of metrics

Reply:
we need to make sure that - we aren't leaking memory from `volumesToUnmount` map. Are we ever removing elements from this map? I don't see it.

Reply:
Yup, will update and perform the delete.

Reply:
You can find the delete here:
https://github.com/kubernetes/kubernetes/pull/57766/files#diff-d905eaaa258ac0c788ec26e31079ce60R371